e2e/generatorForm.playwright.ts

```diff
@@ -228,4 +254,41 @@ test.describe("Generator page functionality", () => {
 
     await expect(page.locator("[name=generatedJson]")).not.toBeVisible();
   });
+
+  it("Remembers states and values", async ({ page }) => {
+    await page.goto("/generator");
+    await fillEssentialFieldsWithDefaults(page);
+    await fillUserFacingFieldsWithDefaults(page);
+    await fillTechnicalFields(page);
+
+    // Cause field error state for empty clientId.
+    await page.locator("[name=clientId]").fill("");
+
+    // Switch page and go back.
+    await page.locator(".openValidatorPage").click();
+    await page.locator(".openGeneratorPage").click();
+
+    // Expect values to have remained the same.
+    await expect(page.locator("[name=clientId]")).toHaveValue("");
+    await expect(page.locator("[name=clientName]")).toHaveValue(
+      DEFAULT_CLIENT_NAME
+    );
+    await expect(page.locator("[name=clientUri]")).toHaveValue(
+      DEFAULT_CLIENT_HOMEPAGE
+    );
+    await expect(page.locator(`[name="redirectUris.0"]`)).toHaveValue(
+      `${DEFAULT_CLIENT_REDIRECT_URI}1`
+    );
+    await expectUserFacingFieldsDefaults(page);
+    await expectTechnicalFieldsDefaults(page);
+
+    // Expect clientId error state to have remained.
+    expect(
+      await page.locator(`.MuiFormHelperText-root.Mui-error`).count()
```

Comment:
This would be a good element to add a `data-testid` attribute to, potentially

Reply:
Changed in https://github.com/inrupt/client-identifier-helper/pull/185/commits/d5425f1bbecc012fc7159de5ba89d7279540550d.